qiita_db/metadata_template.py

```diff
@@ -1073,6 +1073,33 @@ def to_file(self, fp, samples=None):
                 values.insert(0, sid)
                 f.write("%s\n" % '\t'.join(values))
 
+    def to_dataframe(self):
+        conn_handler = SQLConnectionHandler()
+        cols = get_table_cols(self._table, conn_handler)
+        dyncols = get_table_cols(self._table_name(self._id), conn_handler)
+        # remove sample_id from dyncols so not repeated
+        dyncols.remove('sample_id')
+        # Get all metadata for the template
+        sql = """SELECT {0}, {1} FROM qiita.{2} req
+            INNER JOIN qiita.{3} dyn on req.sample_id = dyn.sample_id
+            WHERE req.{4} = %s""".format(
+                ", ".join("req.%s" % c for c in cols),
+                ", ".join("dyn.%s" % d for d in dyncols),
+                self._table, self._table_name(self._id), self._id_column)
+        meta, types = conn_handler.execute_fetchall(sql, [self._id],
+                                                    return_types=True)
+        cols = cols + dyncols
+
+        # Create the dataframe and clean it up a bit
+        df = pd.DataFrame((list(x) for x in meta), columns=cols)
+        df.set_index('sample_id', inplace=True, drop=True)
```

Comment:
I think you can set the index in the [DataFrame constructor](http://pandas.pydata.org/pandas-docs/dev/generated/pandas.DataFrame.html) right?